network/src/main/java/bisq/network/NetworkService.java

```diff
@@ -208,12 +208,13 @@ public CompletableFuture<Boolean> bootstrapToNetwork() {
         log.info("bootstrapToNetwork");
         String nodeId = Node.DEFAULT;
         return serviceNodesByTransport.bootstrapToNetwork(getDefaultPortByTransport(), nodeId)
-                .whenComplete((result, throwable) -> {
+                .thenApply(result -> { // We use thenApply to return a future that completes after this function
                     // If networkNode has not been created we create now one with the default pubKey (default keyId)
                     // and persist it.
-                    if (throwable == null) {
+                    if (result == false) {
```

Comment:
Shouldn't it be `result==true`?

Reply:
Comment says _"If networkNode has not been created..."_ so my understanding is in that case `result` is `false`?

Reply:
Hm... dont remember exaclty. Its a bit unclear and should be improved. I think it was meaning that if the network was not initialized already (I guess we call it just in case and after once initialized it returns fast) we need to persist the keypair. Best to put a breakpoint in to see what the value of result is and if its called repeatedly...

Reply:
The `bootstrapToNetwork` method below seems to return always true if successful.

Reply:
This specific line (`if (result == false)`) is called only once, even when using all 3 networks, with `result` being `true`. So the if condition is never true.

If I change it to `if (result == true)`, persisting the network in the very next line fails with
```
NetworkId for default must be present
```

This makes sense, because bootstrapping != initializing the network. The network ID is only available after the networkService is initialized, but here we only do bootstrapping.

NetworkService.initialize is relatively long, because it only finishes when _all_ enabled networks are up and running.

Would it make sense then to remove the whole `thenApply` block?

```
return serviceNodesByTransport.bootstrapToNetwork(getDefaultPortByTransport(), nodeId)
        .thenApply(result -> { // We use thenApply to return a future that completes after this function
            // If networkNode has not been created we create now one with the default pubKey (default keyId)
            // and persist it.
            if (result == false) {
                persistNetworkId(nodeId, keyPairService.getDefaultPubKey());
            }
            return result;
        });
```

Because its not executed (boolean check is called only once with result = true). And persisting the network would fail because right after bootstrapping the network is too early (it would work after network initialize).

Reply:
I checked out your PR and looked into that. The `if (result == false) ` check is wrong as we always return true from `serviceNodesByTransport.bootstrapToNetwork`. That API is questionable in itself and would require more refactoring work...